antimarkdown/nodes.py

```diff
@@ -208,115 +202,106 @@ class LI(Block):
     def text(self):
         li_env = self.blackboard.setdefault('li-nested-block', [])
         li_env.append(False)
-        li = self.blackboard.get('li-style', [u'*   '])[-1]
-        if hasattr(li, 'next'):
-            li = li.next()
+        li = self.blackboard.get('li-style', ['*   '])[-1]
+        if inspect.isgenerator(li):
+            li = next(li)
         text = whitespace(eltext(self.el.text)).lstrip()
 
-        lines = newlines(u''.join(u'\n' + unicode(node)
-                                  if isinstance(node, Block) else unicode(node)
+        lines = newlines(''.join('\n' + str(node)
+                                  if isinstance(node, Block) else str(node)
```

Comment:
The indentation here and in the next line are one space too many after dropping the `u` wart from the joining string.